workspaces/src/rpc/client.rs

```diff
@@ -332,6 +350,147 @@ impl Client {
+        let resp = self
+            .rpc_client
+            .call(
+                methods::EXPERIMENTAL_changes::RpcStateChangesInBlockByTypeRequest {
+                    block_reference,
+                    state_changes_request,
+                },
+            )
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn check_tx(
+        &self,
+        signed_transaction: SignedTransaction,
+    ) -> Result<RpcBroadcastTxSyncResponse> {
+        let resp = self
+            .rpc_client
+            .call(methods::EXPERIMENTAL_check_tx::RpcCheckTxRequest { signed_transaction })
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn genesis_config(&self) -> Result<GenesisConfig> {
+        let resp = self
+            .rpc_client
+            .call(methods::EXPERIMENTAL_genesis_config::RpcGenesisConfigRequest)
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn protocol_config(
+        &self,
+        block_reference: BlockReference,
+    ) -> Result<ProtocolConfigView> {
+        let resp = self
+            .rpc_client
+            .call(
+                methods::EXPERIMENTAL_protocol_config::RpcProtocolConfigRequest { block_reference },
+            )
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn receipt(&self, receipt_reference: ReceiptReference) -> Result<ReceiptView> {
+        let resp = self
+            .rpc_client
+            .call(methods::EXPERIMENTAL_receipt::RpcReceiptRequest { receipt_reference })
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn tx_status(
+        &self,
+        transaction_info: TransactionInfo,
+    ) -> Result<FinalExecutionOutcomeWithReceiptView> {
+        let resp = self
+            .rpc_client
+            .call(methods::EXPERIMENTAL_tx_status::RpcTransactionStatusRequest { transaction_info })
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn validators_ordered(
+        &self,
+        block_id: MaybeBlockId,
+    ) -> Result<Vec<ValidatorStakeView>> {
+        let resp = self
+            .rpc_client
+            .call(
+                methods::EXPERIMENTAL_validators_ordered::RpcValidatorsOrderedRequest { block_id },
+            )
+            .await
+            .map_err(|e| RpcErrorCode::QueryFailure.custom(e))?;
+        Ok(resp)
+    }
+
+    pub(crate) async fn signed_transaction<U: serde::Serialize>(
+        &self,
+        contract_id: &AccountId,
+        signer: &InMemorySigner,
+        func_name: String,
+        func_args: U,
+    ) -> Result<SignedTransaction> {
```

Comment:
Where did this method come from? This looks like `send_tx_and_retry` to me (it is already implemented above, so I don't see any reason for this method to exist)

Reply:
I needed to access the `InMemorySigner::inner` method and also the `fetch_tx_nonce` which are private outside the workspace.
I found no clean way to construct the `SignedTransaction` on the examples side and added this convenience method.

Reply:
TL;DR: Let's remove `check_tx` and `signed_transaction`. `check_tx` is going to be removed from nearcore soon, and it does not make any sense in the context of workspaces-rs to have this method anyway.

It would be better to introduce a method to get a signed transaction after construction it via various methods - there are a number of helpers here that use `.transact().await` at the end, where we could introduce a new `.prepare_signed_transaction().await`, but it I don't see a demand for that, so we should not spend time on it right now.